Tasks/Gradle/sonarqube-common/sonarqube-rest.ts

```diff
@@ -0,0 +1,193 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
+
+import Q = require('q');
+import url = require('url');
+import http = require('http');
+import {IncomingMessage} from 'http';
+import https = require('https');
+
+import tl = require('../testlib/tasklib-wrapper');
+
+import {TaskReport} from  './taskreport';
+import {RestResponse} from  './restresponse';
+import {SonarQubeEndpoint} from './sonarqube-common';
+import sqCommon = require('./sonarqube-common');
+
+// Repeated query the server to determine if the task has finished.
+// Returns true if the task is finished, or false if the timeout was exceeded without a positive answer from the server.
+// Timeout and delay are in seconds, defaulting to 60 and 1 respectively if not specified
```

Comment:
To change: add a comment to document rejection behaviour (i.e. promise rejects if there is a problem with the request, or if the response code is not 200).